makefile

```diff
@@ -6,72 +6,129 @@ MAKEFLAGS += --warn-undefined-variables
 SHELL := /bin/bash
 .SHELLFLAGS := -eu -o pipefail
 .DEFAULT_GOAL := build
+.PHONY: build ship test help
 
 MANTA_LOGIN ?= triton_mysql
 MANTA_ROLE ?= triton_mysql
 MANTA_POLICY ?= triton_mysql
 
-build:
-	docker-compose -p my -f local-compose.yml build
-
-ship:
-	docker tag my_mysql autopilotpattern/mysql
-	docker push autopilotpattern/mysql
-
-# -------------------------------------------------------
-# testing
+## Display this help message
+help:
+	@awk '/^##.*$$/,/[a-zA-Z_-]+:/' $(MAKEFILE_LIST) | awk '!(NR%2){print $$0p}{p=$$0}' | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}' | sort
 
-DOCKER_CERT_PATH ?=
-DOCKER_HOST ?=
-DOCKER_TLS_VERIFY ?=
-LOG_LEVEL ?= DEBUG
+# ------------------------------------------------
+# Container builds
 
-ifeq ($(DOCKER_CERT_PATH),)
-	DOCKER_CTX := -v /var/run/docker.sock:/var/run/docker.sock -e COMPOSE_FILE=local-compose.yml \
+# we get these from shippable if available, otherwise from git
+COMMIT ?= $(shell git rev-parse --short HEAD)
+BRANCH ?= $(shell git rev-parse --abbrev-ref HEAD)
+TAG := $(BRANCH)-$(COMMIT)
 
-else
-	DOCKER_CTX := -e DOCKER_TLS_VERIFY=1 -e DOCKER_CERT_PATH=$(DOCKER_CERT_PATH:$(HOME)%=%) -e DOCKER_HOST=$(DOCKER_HOST)
-endif
-
-cleanup:
-	$(call check_var, SDC_ACCOUNT, Required to cleanup Manta.)
-	-mrm -r /${SDC_ACCOUNT}/stor/triton-mysql/
-	mmkdir /${SDC_ACCOUNT}/stor/triton-mysql
-	mchmod -- +triton_mysql /${SDC_ACCOUNT}/stor/triton-mysql
+## Builds the application container image
+build:
+	docker build -t=autopilotpattern/mysql:$(TAG) .
 
-build-test:
-	docker build -f tests/Dockerfile -t="test" .
-
-# Run tests by running the test container. Currently only runs locally
-# but takes your DOCKER environment vars to use as the test runner's
-# environment (ex. the test runner runs locally but starts containers
-# on Triton if you're pointed to Triton)
-TEST_RUN := python -m trace
-ifeq ($(TRACE),)
-	TEST_RUN := python
-endif
-
-test:
-	unset DOCKER_HOST \
-	&& unset DOCKER_CERT_PATH \
-	&& unset DOCKER_TLS_VERIFY \
-	&& docker run --rm $(DOCKER_CTX) \
-		-e LOG_LEVEL=$(LOG_LEVEL) \
-		-e COMPOSE_HTTP_TIMEOUT=300 \
-		--env-file=_env \
-		-v ${HOME}/.triton:/.triton \
-		-v ${HOME}/src/autopilotpattern/testing/testcases.py:/usr/lib/python2.7/site-packages/testcases.py \
-		-v $(shell pwd)/tests/tests.py:/src/tests.py \
-		-w /src test $(TEST_RUN) tests.py
+## Pushes the application container image to the Docker Hub
+ship:
+	docker push autopilotpattern/mysql:$(TAG)
```

Comment:
@misterbisson should we be simultaneously pushing the `:latest` tag here, do you think?